### PR TITLE
Sync

### LINKED ISSUE #12: pre-commit hooks
Just an idea
https://github.com/antonbabenko/pre-commit-terraform

### LINKED ISSUE #52: Sync with template and update to use 0.6.0 box

### PR DESCRIPTION
**Pull request by dangernil**
## Closes/fixes/resolves issue(s)?
Closes #52 
Closes #12 

## What was added/changed/fixed?
Synced with template and box version 0.6.0

## Checklist (after created PR)
- [x] Added reviewers
- [x] Added assignee(s)
- [x] Added label(s)
- [x] Added to project
- [x] Added to milestone

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +50,12 @@
 - Initial example
 - Changelog #7
 
+## [0.1.0 UNRELEASED]
+
+## Changed
+
+- updated to fit hashistack v0.6.0
+
```
Comment (pdmthorsrud) [inline CHANGELOG.md]: Seems like there is another `0.1.0` above, isn't it?

Comment (claesgill) [inline CHANGELOG.md:58]: As @pdmthorsrud says, line 53-61 should be removed. It's changelog from the template: https://github.com/fredrikhgrelland/vagrant-hashistack-template/blob/master/CHANGELOG.md

Comment (claesgill) [reply CHANGELOG.md]: That's right, line 53-61 should be removed. It's changelog from the template: https://github.com/fredrikhgrelland/vagrant-hashistack-template/blob/master/CHANGELOG.md

File: Makefile [hunk 2]
```diff
@@ -80,7 +78,8 @@ proxy:
 	consul intention create -token=master minio-local minio
 	consul connect proxy -token master -service minio-local -upstream minio:9000 -log-level debug
 
-pre-commit: check_for_docker_binary
+pre-commit: check_for_docker_binary check_for_terraform_binary
+
```
Comment (claesgill) [inline Makefile]:
```suggestion
```
Sorry - could not unsee it :sweat_smile:

Comment (claesgill) [reply Makefile:82]: :sweat_smile:

**Comment by claesgill** [comment]
QUESTION: @dangernil @pdmthorsrud @zhenik 
Since we do not use the `template_example`, what stops us in deleting it?

**Comment by pdmthorsrud** [comment]
> QUESTION: @dangernil @pdmthorsrud @zhenik
> Since we do not use the `template_example`, what stops us in deleting it?

Very good question. I guess nothing stops us from deleting it. Only reason we have it is to make sure everything works as expected in the template. I don't like having to remove files or folders whenever you use our template though, because it creates more uncertainty when you sync with newer versions of template (which will try and add back the template_example).

**Review by claesgill — DISMISSED** [review]
LGTM 🚀

**Comment by claesgill** [comment]
> > QUESTION: @dangernil @pdmthorsrud @zhenik
> > Since we do not use the `template_example`, what stops us in deleting it?
> 
> Very good question. I guess nothing stops us from deleting it. Only reason we have it is to make sure everything works as expected in the template. I don't like having to remove files or folders whenever you use our template though, because it creates more uncertainty when you sync with newer versions of template (which will try and add back the template_example).

I get your concern. However, if we decide to delete it completely and later on want to sync with the template, the sync should only add a new folder `template_example` with no conflicts etc. We can just accept the `template_example` folder being added and delete it yet again.

If we don't delete, we need to fix conflicts (in `template_example`) that are not even used every time we sync. I guess there's not big changes in that folder in each sync, but you know.

It's a tricky question, so I'm not sure what is best. Just thinking out loud here, but maybe decoupling `template_example` from template could be a solution in the long run?

**Comment by zhenik** [comment]
I actually like the idea to delete `template_example` directory. This directory relevant only to the template but not modules

**Comment by pdmthorsrud** [comment]
> I actually like the idea to delete `template_example` directory. This directory relevant only to the template but not modules

Fair enough, if you guys like it, I'm down to try :D

File: CHANGELOG.md [hunk 3]
```diff
@@ -49,10 +50,6 @@
 - Initial example
 - Changelog #7
 
-### Fixed
-
-- Remove hardcoded nomad provider #6
-
```
Comment (pdmthorsrud) [inline CHANGELOG.md]: This shouldn't be removed, it's a part of the template's changelog, isn't it? (I might be wrong here)

Comment (dangernil) [reply CHANGELOG.md]: Yes, this should have been part of the minio changelog. It's on its way back

Comment (pdmthorsrud) [reply CHANGELOG.md]: Nice :D Think that's everything for me, rest looks good!

File: CHANGELOG.md [hunk 4]
```diff
@@ -49,7 +50,7 @@
 - Initial example
 - Changelog #7
 
-### Fixed
+###Fixed
```
Comment (zhenik) [inline CHANGELOG.md]:
```suggestion
### Fixed
```

File: dev/ansible/03_run_tests.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 - name: set variable
   set_fact:
-    consul_token: "master"
+    consul_token: "{{ lookup('env', 'consul_master_token') }}"
```
Comment (zhenik) [inline dev/ansible/03_run_tests.yml:3]: I would recommend staying consistent.
```suggestion
    consul_token: "{{ lookup('env', 'consul_master_token') }}"
    vault_addr: "{{ lookup('env', 'VAULT_ADDR') }}"
    vault_token: "{{ lookup('env', 'vault_master_token') }}"
```

**Comment by claesgill** [comment]
> > I actually like the idea to delete `template_example` directory. This directory relevant only to the template but not modules
> 
> Fair enough, if you guys like it, I'm down to try :D

I hope you like it as well, @pdmthorsrud :sweat_smile: 

But then i think that @dangernil should remove the `template_example` in this PR. Do you guys agree, @dangernil @pdmthorsrud @zhenik ?

File: dev/ansible/03_run_tests.yml [hunk 6]
```diff
@@ -79,11 +79,17 @@
 - name: Register access key
   shell:
     cmd: vault kv get -field='access_key' secret/minio
+  environment:
+    VAULT_ADDR: "{{ lookup('env', 'VAULT_ADDR') }}"
+    VAULT_TOKEN: "{{ lookup('env', 'vault_master_token') }}"
```
Comment (zhenik) [inline dev/ansible/03_run_tests.yml]: Consistency. 
Either use `set_facts` or `lookup`

File: dev/ansible/03_run_tests.yml [hunk 7]
```diff
@@ -79,11 +79,17 @@
...
   register: access_key
 
 - name: Register secret key
   shell:
     cmd: vault kv get -field='secret_key' secret/minio
+  environment:
+    VAULT_ADDR: "{{ lookup('env', 'VAULT_ADDR') }}"
+    VAULT_TOKEN: "{{ lookup('env', 'vault_master_token') }}"
```
Comment (zhenik) [inline dev/ansible/03_run_tests.yml]: Same comment as above

File: template_example/example/standalone/providers.tf [hunk 8]
```diff
@@ -10,3 +10,18 @@ provider "nomad" {
...
+terraform {
+  required_providers {
+    vault = {
+      version = ">=2.13.0"
+    }
+    nomad = {
+      version = ">=1.4.0"
+    }
+  }
+}
```
Comment (zhenik) [inline template_example/example/standalone/providers.tf]:
```suggestion
provider "vault" {
  address = "http://127.0.0.1:8200"
  token = "master"
}
terraform {
  required_providers {
    vault = {
      version = ">=2.14.0"
    }
    nomad = {
      version = ">=1.4.9"
    }
  }
}
```

**Review by zhenik — CHANGES_REQUESTED** [review]

**Comment by zhenik** [comment]
> > > I actually like the idea to delete `template_example` directory. This directory relevant only to the template but not modules
> > 
> > 
> > Fair enough, if you guys like it, I'm down to try :D
> 
> I hope you like it as well, @pdmthorsrud 😅
> 
> But then i think that @dangernil should remove the `template_example` in this PR. Do you guys agree, @dangernil @pdmthorsrud @zhenik ?

I think it is worth it to try

File: example/main.tf [hunk 9]
```diff
@@ -1,3 +1,8 @@
+provider "vault" {
+  address = "http://127.0.0.1:8200"
+  token = "master"
+}
+
```
Comment (zhenik) [inline example/main.tf]: Better move this block under providers.tf .
Also, could you set up the providers version explicitly

**Review by zhenik — CHANGES_REQUESTED** [review]

File: example/providers.tf [hunk 10]
```diff
@@ -4,9 +4,28 @@ data "vault_generic_secret" "nomad_secret_id" {
   path  = "nomad/creds/write"
 }
 
+data "vault_generic_secret" "minio_secrets" {
+  path  = "secret/minio"
+}
+
```
Comment (zhenik) [inline example/providers.tf]: I would recommend leave this block in main.tf , due to this block lies in minio context

**Review by zhenik — CHANGES_REQUESTED** [review]

**Review by zhenik — APPROVED** [review]
LGTM

**Review by claesgill — APPROVED** [review]
LGTM

**Pull request merged**